### PR TITLE
1 smoke effect

### LINKED ISSUE #1: Smoke effect under the player when jumping
Make smoke using particle effect

### PR DESCRIPTION
**Pull request by seannam1218**
Made a particle effect that plays when player jumps.
There is a bug where the number of available jumps refreshes before the character lands on the ground.
When said bug is fixed, the smoke effect should only play from the ground and not in the air.

P.S.
I tried getting particle effect to work when landing instead of jumping, as this would also make sense. However, the raycast isGrounded detection seems to be not working correctly, which would explain the bug explained above.

### CODE REVIEW

File: Assets/Scripts/Player.cs [hunk 1]
```diff
@@ -7,9 +7,12 @@ public class Player : MonoBehaviour {
 	[SerializeField] float jumpSpeed = 7f;
 	[SerializeField] int maxJumps = 2;
 	[SerializeField] int jumpsAvailable;
+	[SerializeField] private ParticleSystem dust;
+	bool isNotYetLanded;
```
Comment (asim-shrestha) [inline Assets/Scripts/Player.cs]: remove?

File: Assets/Scripts/Player.cs [hunk 2]
```diff
@@ -58,6 +61,9 @@ public class Player : MonoBehaviour {
 		} else if (Input.GetButtonDown("Jump")) {
 			Vector2 jumpVelocity = new Vector2(rb.velocity.x, jumpSpeed);
 			rb.velocity = jumpVelocity;
+			if (jumpsAvailable >= 2) {
```
Comment (asim-shrestha) [inline Assets/Scripts/Player.cs]: should never be greater so could just be == (Not a big deal)

**Comment by asim-shrestha** [comment]
yeah mb with the raycasting, still needs some fine tuning. I think i might have made it better in my branch

File: Assets/Scripts/GroundCheck.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+		notYetLanded = true;
+	}
+
+	// Update is called once per frame
+	void Update()
+	{
+
+	}
+
+	void OnTriggerEnter(Collider other) {
```
Comment (asim-shrestha) [inline Assets/Scripts/GroundCheck.cs]: Probably looking for OnTriggerEnter2D?

Comment (seannam1218) [reply Assets/Scripts/GroundCheck.cs]: I was playing around with this and forgot to delete this file. The newest push deleted this file.

File: Assets/Scripts/Player.cs (hunk 2 above)
Comment (seannam1218) [reply Assets/Scripts/Player.cs]: fixed to jumpsAvailable == maxJumps

File: Assets/Scripts/Player.cs (hunk 1 above)
Comment (seannam1218) [reply Assets/Scripts/Player.cs]: removed the variable isNotYetLanded. Need the ParticleSystem tho, just for clarification.

File: Assets/Scripts/GroundCheck.cs (hunk 3 above)
Comment (seannam1218) [reply Assets/Scripts/GroundCheck.cs]: This functionality is redundant if the raycast ground check works

**Review by asim-shrestha — APPROVED** [review]

**Pull request merged**